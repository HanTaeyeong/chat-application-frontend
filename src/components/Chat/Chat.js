import React, { useState, useEffect } from 'react';
import queryString from 'query-string'
import io from 'socket.io-client'

import './Chat.scss'

import InfoBar from '../InfoBar/InfoBar'
import Input from '../Input/Input'

let socket;

const ENDPOINT = 'http://44.202.97.232:5000'
const extraHeaders = {
  // Authorization: "Bearer authorization_token_here"
}

export default function Chat() {
  const [value, setValue] = useState({ name: '', room: '' })
  const [message, setMessage] = useState('')
  const [messages, setMessages] = useState([])
  useEffect(() => {
    const { name, room } = queryString.parse(window.location.search)

    //const socket = openSocket('http://localhost:8000', , {transports: ['websocket']});
    socket = io(ENDPOINT)

    setValue({ name, room })

    socket.emit('join', { name, room })
    return () => {
      socket.emit('disconnect')
      socket.off()
    }
  }, [ENDPOINT, window.location.search])

  useEffect(() => {
    socket.on('message', (message) => {
      setMessages([...messages, message])
      console.log(value)
    })

  }, [messages])

  const sendMessage = (e) => {
    e.preventDefault()

    const { message } = value;
    if (message) {
      socket.emit('sendMessage', message)
      setMessage('')
    }
  }

  return <div className='chat-outer-container'>
    <div className='chat-inner-container'>
      <InfoBar room={value.room} />
      <Input/>
    </div>
    Chat
  </div>
}
