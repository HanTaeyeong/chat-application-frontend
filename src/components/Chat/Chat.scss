.chat-outer-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  background-color: #1a1a1d;
}

.chat-inner-container {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: #ffffff;
  border-radius: 8px;
  height: 60%;
  width: 35%;
}

@media (min-width: 320px) and (max-width: 480px) {
  .chat-outer-container {
    height: 100%;
  }

  .chat-inner-container {
    width: 100%;
    height: 100%;
  }
}

@media (min-width: 480px) and (max-width: 1200px) {
  .chat-inner-container {
    width: 60%;
  }
}
