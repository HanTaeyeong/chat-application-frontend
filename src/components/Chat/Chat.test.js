import React from 'react'
import Chat from './Chat'

import {create} from 'react-test-renderer'


describe('Chat Component Test', () => {
    test('render test', () => {   
        let app=create(<Chat />) 
        expect(app.toJSON()).toMatchSnapshot()
    })
})