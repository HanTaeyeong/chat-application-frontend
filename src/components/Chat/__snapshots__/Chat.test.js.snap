// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Component Test render test 1`] = `
<div
  className="chat-outer-container"
>
  <div
    className="chat-inner-container"
  >
    <div
      className="infobar-outer-container"
    >
      <div
        className="infobar-left-inner-container"
      >
        <img
          alt="online-icon"
          className="infobar-online-icon"
          src="onlineIcon.png"
        />
        <h3>
           
           
        </h3>
      </div>
      <h5>
        0
      </h5>
      <div
        className="infobar-right-inner-container"
      >
        <a
          href="/"
        >
          <img
            alt="close-icon"
            src="closeIcon.png"
          />
        </a>
      </div>
    </div>
    <div
      className="react-scroll-to-bottom--css-pyfvl-79elbk messages"
    >
      <div
        className="react-scroll-to-bottom--css-pyfvl-1n7m0yu"
      />
    </div>
    <form
      className="form"
    >
      <input
        className="input"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Type a message"
        type="text"
        value=""
      />
      <button
        className="send-button"
        onClick={[Function]}
      >
        Send
      </button>
    </form>
  </div>
</div>
`;
